users/views.py

```diff
@@ -1,3 +1,190 @@
+            return Response({'message': 'Invalid token.'}, status=status.HTTP_400_BAD_REQUEST)
+
+        if user.is_active:
+            return Response({'message': 'Account already activated.'}, status=status.HTTP_400_BAD_REQUEST)
+
+        # Activate the user, email is confirmed
+        user.is_active = True
+        user.save()
+
+        return Response({'message': 'Account activated successfully.'}, status=status.HTTP_200_OK)
+
+
+class LogoutView(APIView):
+    """
+    It is recommended to provide a way for users to log out of an application in a REST API.
+    Logging out terminates the user's session and invalidates their authentication token. This helps to ensure the
+    security of the user's account and data, especially if the user is accessing the application from a public or
+    shared device.
+
+    To access a view that requires authentication, you need to include the authentication credentials in the request
+    headers.
+    One way to do this is to include the token key in the Authorization header of the request.
+
+    For example, to test this endpoint, you should include the token key in the `Authorization` header like this :
+    `Authorization: Token 9944b09199c62bcf9418ad846dd0e4bbdfc6ee4b`
+
+    (In swagger click on lock icon and add value like this: `Token 9944b09199c62bcf9418ad846dd0e4bbdfc6ee4b`)
+    """
+    authentication_classes = [TokenAuthentication]
+    permission_classes = [IsAuthenticated]
+
+    def post(self, request, ):
+        """
+        In the `post` method, we simply delete the token associated with the current user and return a success message.
+        If the user was not authenticated, we return an error message with a `400 status code`.
+        """
+        try:
+            token = request.auth
+            token.delete()
+            return Response({"message": "You have been logged out."}, status=status.HTTP_200_OK)
+        except Exception as e:
+            return Response({"error": str(e)}, status=status.HTTP_400_BAD_REQUEST)
+
+
+class PasswordResetView(GenericAPIView):
+    serializer_class = serializers.PasswordResetSerializer
+
+    @method_decorator(csrf_protect)
+    def post(self, request):
+        serializer = self.serializer_class(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        user = serializer.validated_data['user']
+
+        # Delete any existing tokens for the user
+        Token.objects.filter(user=user).delete()
+        # Create a new token for the user
+        token = Token.objects.create(user=user)
+
+        # Send password reset email
+        email.SendEmail.send_password_reset(request, user, token)
+
+        return Response({'detail': 'Password reset email sent.'}, status=status.HTTP_200_OK)
+
+
+class PasswordResetConfirmView(GenericAPIView):
+    serializer_class = serializers.PasswordResetConfirmSerializer
+
+    def post(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data)
```

Comment:
And we can send the token within the context here.

```suggestion
        serializer = self.get_serializer(data=request.data, context={'token':kwargs.get('token')})
```

Reply:
Nice, thank you.